### DISCUSSION TITLE
time.time() is off by 30 years

### DISCUSSION THREAD

**Discussion by dannybr** [micropython/micropython#14114] [ESP32] [answered]
after running ntp, the time.localtime() is fine, but time.time() is off by some 30 years!

**Comment by robert-hh** [comment]
time.time() returns the seconds since the EPOCH, which is the 1.1.2000 on an ESP32. That's returned by time.gmtime()[0]. Testing it on an ESP32, it returns something like 764_005_006, which is a reasonable number right. For getting a time since 1.1.1970, you have to add 946_684_800.

**Comment by Josverl** [comment, marked as answer]
The majority of MCU ports of Micropython uses a different epoch (start datetime) than CPython. 

> epoch of 2000-01-01 00:00:00 UTC. Epoch year may be determined with gmtime(0)[0].

See https://docs.micropython.org/en/latest/library/time.html

**Comment by dannybr** [comment]
RTFM, but then again it’s our 600 paged :-)
IMHO, this should be in CPYthon Micropython difference section.

thanks,
	danny

**Reply by Josverl** [reply]
Just don't print it, use the micropython stubs instead
